packages/agora-rtc-react/src/hooks/tracks.ts

```diff
@@ -349,3 +355,109 @@ export function useRemoteVideoTracks(
   }, [isConnected, resolvedClient, users]);
   return tracks;
 }
+
+/**
+ * a hook can create a local video track
+ * unpublish track on unmount.
+ */
+export function useLocalVideoTrack(
+  ready = true,
+  client?: IAgoraRTCClient,
+): ICameraVideoTrack | null {
+  const isConnected = useIsConnected(client);
+  const [track, setTrack] = useState<ICameraVideoTrack | null>(null);
+
+  useAsyncEffect(async () => {
+    if (isConnected && ready && !track) {
+      setTrack(await AgoraRTC.createCameraVideoTrack());
+    }
+    if (!isConnected) {
+      setTrack(null);
+    }
+  }, [isConnected, ready]);
+  return track;
+}
+
+/**
+ * a hook can create a local audio track
+ * close track on unmount.
+ */
+export function useLocalAudioTrack(
+  ready = true,
+  audioTrackConfig: MicrophoneAudioTrackInitConfig = { ANS: true, AEC: true },
```

Comment:
The config object is not listened below in the effect, so any changes to this object will not trigger a recreation to the track. We'd better document that in the doc comments.

Reply:
Agreed, if users want more configurable local track creation hooks in the future, more work may be needed. 
Thanks for your suggestions.
My current design idea is to only use it as a one-time creation. As long as the client is not disconnected, even if a     re-render occurs, I will not recreate a new track either.